### PR TITLE
SAPI-347 : feature/integrate cache broker

### PR DESCRIPTION
**Pull request by bpathak-ch**
* Further client and broker implementation for frontend service
* Unit test updates
* logger updates
* removed unused files

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"github.com/companieshouse/chs-streaming-api-frontend/logger"
+	"github.com/companieshouse/chs.go/log"
+	"net/http"
+	"os"
+	"sync"
+
+	//"sync"
```
Comment (dlgroves-ch) [inline client/client.go]: Please remove commented out code

File: client/client.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+func (c *Client) Connect() {
+	resp, _ := c.client.Get(c.baseurl)
+	body := resp.Body
+	reader := bufio.NewReader(body)
+	go c.loop(reader)
+}
+
+func (c *Client) loop(reader *bufio.Reader) {
+	//
```
Comment (dlgroves-ch) [inline client/client.go]: Please remove commented out code.

File: handlers/stream.go [hunk 3]
```diff
@@ -78,9 +83,22 @@ func (st Streaming) ProcessHTTP(w http.ResponseWriter, req *http.Request) {
...
 			w.(http.Flusher).Flush()
+		case msg := <-subscription:
+			st.logger.InfoR(req, "User connected")
+			_, _ = w.Write([]byte(msg))
+			w.(http.Flusher).Flush()
+			if st.wg != nil {
+				st.wg.Done()
+			}
+		case <-req.Context().Done():
+			st.logger.InfoR(req, "User disconnecting")
```
Comment (dlgroves-ch) [inline handlers/stream.go]: This event will only be received when the user has disconnected; should be "user disconnected"

File: client/client.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		line, err := reader.ReadBytes('\n')
+		if err != nil {
+			c.logger.Error(err, log.Data{})
+			fmt.Fprintf(os.Stderr, "error during resp.Body read:%s\n", err)
+			continue
+		}
+
+		result := &Result{}
+
+		json.Unmarshal(line, result)
```
Comment (dlgroves-ch) [inline client/client.go]: We shouldn't be doing anything with the JSON that we're consuming here; we should be outputting what we receive from chs-streaming-api-cache directly to all connected users.

**Review by dlgroves-ch — CHANGES_REQUESTED** [review]

**Review by dlgroves-ch — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Pull request merged**